### PR TITLE
Early Access: Self Serve Custom Upstreams

### PR DESCRIPTION
**Pull request by rachelwhitton**
Re: [AL-1073](https://getpantheon.atlassian.net/browse/AL-1073) & [AL-1132](https://getpantheon.atlassian.net/browse/AL-1132)

## Effect
PR includes the following changes:
- Repurpose create-custom-upstream.md for early access doc
- Add example usage to Terminus manual: Switch Upstreams

## Remaining Work
- [x] Verify eligibility
- [x] Peer review with Alchemy

### CODE REVIEW

File: source/_docs/create-custom-upstream.md [hunk 1]
```diff
@@ -1,19 +1,78 @@
...
-categories: []
+earlyaccess: true
+earlynote: The documentation on this page discusses features and options that are not available across the entire platform.
 ---
+Pantheon is rolling out a new self-serve feature to support creating Custom Upstreams in the Organization Dashboard. Once created, members of the organization will be able to create new sites from the common codebase. For an overview of this feature, see [Introduction to Custom Upstreams](/docs/custom-upstream).
 
-The following steps will help you get a Custom Upstream ready for prime time within your organization. Once completed, members of the organization will be able to create new sites from the common codebase. For an overview of this feature, see [Introduction to Custom Upstreams](/docs/custom-upstream).
+## Eligibility
+Access to this feature is currently invite only for Agency and Enterprise Organizations. Repositories must be hosted on [GitHub](https://github.com/) or [BitBucket](https://bitbucket.org/). If you would like early access to this feature, or if you need to use an alternate repository hosting provider, please let us know.
+
```
Comment (kimby77) [inline source/_docs/create-custom-upstream.md]: Suggestion for clarity:

"Contact us for early access to this feature. If you need to use an alternate repository hosting provider, let us know in the form."

File: source/_docs/terminus/examples.md [hunk 2]
```diff
@@ -129,6 +130,53 @@ The following command will permanently delete all uncommitted SFTP changes. If y
 <figure><pre id="mass-update-apply"><code class="command bash" data-lang="bash">terminus site:list --format=list | terminus site:mass-update:apply --accept-upstream --updatedb</code></pre></figure>
 </div>
 
+## Switch Upstreams
+Every site has an upstream assigned in order to deliver [one-click updates](/docs/upstream-updates/) in the Pantheon Site Dashboard. Terminus can be used to manage this site level configuration.
+### Move an existing site to a Custom Upstream
+<p class="instruction">If your organization has a <a href="/docs/custom-upstream/">Custom Upstream</a>, you can use Terminus to switch existing sites over to the common codebase:</p>
+<div class="copy-snippet">
+<button class="btn btn-default btn-clippy" data-clipboard-target="#upstream-set">Copy</button>
+<figure><pre id="upstream-set"><code class="command bash" data-lang="bash">terminus site:upstream:set my-site "My Custom Upstream"</code></pre></figure>
```
Comment (kimby77) [inline source/_docs/terminus/examples.md]: Should this call out it's using the site machine name and the upstream machine name?

**Comment by kimby77** [comment]
Should the code snippets here have the "copy" button?

File: source/_docs/terminus/examples.md [hunk 3]
```diff
@@ -129,6 +130,53 @@ The following command will permanently delete all uncommitted SFTP changes. If y
 <figure><pre id="mass-update-apply"><code class="command bash" data-lang="bash">terminus site:list --format=list | terminus site:mass-update:apply --accept-upstream --updatedb</code></pre></figure>
 </div>
 
+## Switch Upstreams
+Every site has an upstream assigned in order to deliver [one-click updates](/docs/upstream-updates/) in the Pantheon Site Dashboard. Terminus can be used to manage this site level configuration.
```
Comment (kimby77) [inline source/_docs/terminus/examples.md]: I know this is still in progress, but this intro paragraph would be a good time to give a few example use cases we know users have from the research, to make this more concrete and help users understand why they might want to do this.

File: source/_docs/terminus/examples.md (hunk 2 above)
Comment (rachelwhitton) [inline source/_docs/terminus/examples.md]: I'm not sure how to refer to site or upstream labels without defining them somewhere. @ronan thoughts? is this necessary?

**Comment by kimby77** [comment]
* This "contact us" should be a more prominent CTA - it's getting lost as just a link and will be easy to overlook.  I'd suggest a button beneath this paragraph.

image

* On the "create a custom upstream" doc I think some screenshots of the UI with maybe some arrow callouts would help orient the user and help them understand what we're referring to. When we talk about something like the "organization dashboard" I've seen in user testing that users may not know what that means, or refer to it with different terminology.  What do you think?

* Love the use case examples in the terminus manual!

* On the example to change a site upstream, there's a few items in the ticket's acceptance criteria not here - it might start to turn this into a lengthy example but do you think we can add these and see how it is? 
** What to do when something unexpected occurs (like unexpected merge conflicts)
** How this works with a Composer workflow (I may have seen some discussion about leaving that out of here?) 
** Safeguards - currently there is the only one to prevent users from switching frameworks.

**Pull request merged**